src/plugins/network/ManageMockResponsePanel.tsx

```diff
@@ -0,0 +1,287 @@
+  routes: Map<RequestId, Route>;
+};
+
+const ColumnSizes = {
+  route: 'flex',
+};
+
+const Columns = {
+  route: {
+    value: 'Route',
+    resizable: false,
+  },
+};
+
+const AddRouteButton = styled(FlexBox)({
+  color: colors.blackAlpha50,
+  alignItems: 'center',
+  padding: 10,
+  flexShrink: 0,
+  whiteSpace: 'nowrap',
+  overflow: 'hidden',
+  textOverflow: 'ellipsis',
+});
+
+const Container = styled(FlexRow)({
+  flex: 1,
+  justifyContent: 'space-around',
+  alignItems: 'stretch',
+});
+
+const LeftPanel = styled(FlexColumn)({
+  flex: 1,
+});
+
+const RightPanel = styled(FlexColumn)({
+  flex: 3,
+  height: '100%',
+});
+
+const TextEllipsis = styled(Text)({
+  overflowX: 'hidden',
+  textOverflow: 'ellipsis',
+  maxWidth: '100%',
+  lineHeight: '18px',
+  paddingTop: 4,
+  display: 'block',
+  whiteSpace: 'nowrap',
+});
+
+const Icon = styled(Glyph)({
+  marginTop: 5,
+  marginRight: 8,
+});
+
+export class ManageMockResponsePanel extends Component<Props, State> {
+  static ContextMenu = styled(ContextMenu)({
+    flex: 1,
+  });
+
+  contextMenuItems() {
+    return [
+      {
+        type: 'separator' as 'separator',
+      },
+      {
+        label: 'Delete',
+        click: this.deleteRoute,
+      },
+    ];
+  }
+
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      ...this.state,
+      selectedIds:
+        this.props.routes !== undefined && this.props.routes.size > 0
+          ? ['0']
+          : [],
+    };
+  }
+
+  deleteRoute = () => {
+    const {selectedIds} = this.state;
+    const selectedId = selectedIds.length === 1 ? selectedIds[0] : null;
+    const routes = this.props.routes;
+    if (selectedId !== null) {
+      routes.delete(selectedId);
+    }
+
+    this.checkDuplicate(routes);
+
+    this.setState(
+      {
+        selectedIds: [],
+      },
+      () => {
+        this.props.handleRoutesChange(routes);
+      },
+    );
+  };
+
+  buildRows = () => {
+    const {routes} = this.props;
+    if (routes) {
+      const rows: TableBodyRow[] = [];
+
+      routes.forEach((route: Route, index: RequestId) => {
+        rows.push(this.buildRow(route, index));
+      });
+      return rows;
+    }
+    return [];
+  };
+
+  buildRow = (route: Route, index: RequestId) => {
+    return {
+      columns: {
+        route: {
+          value: route.isDuplicate ? (
+            <FlexRow>
+              <Icon name="caution-triangle" color={colors.yellow} />
+              <TextEllipsis>{route.requestUrl}</TextEllipsis>
+            </FlexRow>
+          ) : (
+            <TextEllipsis>{route.requestUrl}</TextEllipsis>
+          ),
+        },
+      },
+      key: index,
+    };
+  };
+
+  checkDuplicate = (routes?: Map<RequestId, Route>) => {
+    const duplicateMap: {[key: string]: boolean} = {};
+    if (routes && routes.size > 0) {
+      routes.forEach((r: Route) => {
+        if (duplicateMap[r.method + '|' + r.requestUrl]) {
+          r.isDuplicate = true;
```

Comment:
The fundamental issue here remains that props are mutated. This is incompatible with the React model and will break. I would strongly recommend to make methods like these pure, which is both easier to reason about and will ensure you comply with the React contracts.